specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2022-03-08-preview/Database.json

```diff
@@ -0,0 +1,319 @@
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./common-types.json#/parameters/ServerNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DatabaseNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/Database"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforPostgreSQL/flexibleServers/{serverName}/databases": {
+      "get": {
+        "tags": [
+          "Databases"
+        ],
+        "operationId": "Databases_ListByServer",
+        "x-ms-examples": {
+          "List databases in a server": {
+            "$ref": "./examples/DatabasesListByServer.json"
+          }
+        },
+        "description": "List all the databases in a given server.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./common-types.json#/parameters/ServerNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/DatabaseListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "DatabaseProperties": {
+      "type": "object",
+      "properties": {
+        "charset": {
+          "type": "string",
+          "description": "The charset of the database."
```

Comment:
>description [](http://example.com/codeflow?start=11&length=11)

What are the valid values? Are you allowing anything from some ISO standard, or what?

Reply:
There is a huge list of supported charsets for POSTGRESQL. Since the list is big there is no enum defined to restrict the charset. https://www.postgresql.org/docs/current/multibyte.html#:~:text=The%20character%20set%20support%20in%20PostgreSQL%20allows%20you,%28Extended%20Unix%20Code%29%2C%20UTF-8%2C%20and%20Mule%20internal%20code.

Reply:
Can you at least include some explanation of what is supported, or link to the documentation for that?

Reply:
Added pattern validation